compiler/rustc_error_codes/src/error_codes/E0802.md

```diff
@@ -0,0 +1,41 @@
+An auto trait cannot be added to the bounds of a `dyn Trait` type via
+a pointer cast.
+
+Erroneous code example:
+
+```rust,edition2021,compile_fail,E0802
+let ptr: *const dyn core::any::Any = &();
+_ = ptr as *const (dyn core::any::Any + Send);
+```
+
+Adding an auto trait can make the vtable invalid, potentially causing
+UB in safe code afterwards. For example:
+
+```rust,ignore
```

Comment:
hm, maybe was intended to be
```suggestion
```rust,compile_fail
```

or
```suggestion
```rust,no_run
```

Reply:
Yes indeed.